### DISCUSSION TITLE
Asking about midi data in event of note

### DISCUSSION THREAD

**Discussion by RProduction** [nlaha/godot-midi#30] [Q&A]
Hello, I want to ask.
What event['delta'] and event['data'] stand for? I want to get duration or length of a note. Do i get it by calculating between note on and note off real time or i can use delta at note on?

Thank you

**Comment by nlaha** [comment]
Event delta is the time before the event, not the duration. Ultimately this is a tricky problem to solve, and I've been thinking of adding a MIDI  preprocessing step that calculates absolute times & durations for each event at import. That said, it won't be added anytime soon so calculating it at runtime or before the track plays is the way to go for now.

Event data can be a variety of things, for note-on it's the velocity.

This page might help: https://midimusic.github.io/tech/midispec.html

**Reply by RProduction** [reply]
I see, thank you for the answer. I guess, i will try to wait for the feature. Meanwhile, I will use another approach to get length of midi note.

I am using this to make rhythm game. Currently my long node is determined by fixed length. Probably the approach from the demo will work for long note that i determined in black key note.


Thank you.
